querier.go

```diff
@@ -231,22 +261,27 @@ func (r *postingsReader) selectSingle(m labels.Matcher) Postings {
 		return it
 	}
 
-	// TODO(fabxc): use interface upgrading to provide fast solution
-	// for prefix matches. Tuples are lexicographically sorted.
 	tpls, err := r.index.LabelValues(m.Name())
 	if err != nil {
 		return errPostings{err: err}
 	}
 
 	var res []string
-
-	for i := 0; i < tpls.Len(); i++ {
-		vals, err := tpls.At(i)
+	if pm, ok := m.(*labels.PrefixMatcher); ok {
+		res, err = findTuplesByPrefix(pm, tpls)
 		if err != nil {
 			return errPostings{err: err}
 		}
-		if m.Matches(vals[0]) {
-			res = append(res, vals[0])
+
```

Comment:
Not sure if I understood correctly but I still need to run the code retrieving postings below regardless of whether it's a prefix matcher or not.